### PR TITLE
feat: add updateStartAndEndTimes & roundFee

### LINKED ISSUE #1158: Round Implementation explore size reduction
Explore reducing the bytecode size of the RoundImplementation by grouping update RoundTime functions into 1

### PR DESCRIPTION
**Pull request by bhargavaparoksham**
##### Description

- Reduces the bytecode size of the RoundImplementation by grouping update RoundTime functions into 1
- Adds roundFee

##### Refers/Fixes

fixes #1158

### CODE REVIEW

**Comment by melissa-neira** [comment]
defer to engineers 🐙

File: packages/contracts/contracts/round/RoundImplementation.sol [hunk 1]
```diff
@@ -261,56 +297,44 @@ contract RoundImplementation is AccessControlEnumerable, Initializable {
...
-
-    emit RoundEndTimeUpdated(roundEndTime, newRoundEndTime);
-
-    roundEndTime = newRoundEndTime;
-  }
-
-  /// @notice Update applicationsStartTime (only by ROUND_OPERATOR_ROLE)
-  /// @param newApplicationsStartTime new applicationsStartTime
-  function updateApplicationsStartTime(uint256 newApplicationsStartTime) external roundHasNotEnded onlyRole(ROUND_OPERATOR_ROLE) {
+  function updateStartAndEndTimes(
```
Comment (thelostone-mc) [inline packages/contracts/contracts/round/RoundImplementation.sol:306]: @gravityblast

Comment (gravityblast) [reply packages/contracts/contracts/round/RoundImplementation.sol:306]: LGTM!

**Comment by thelostone-mc** [comment]
Don't merge this until we have the fee + amount

File: packages/contracts/contracts/round/RoundImplementation.sol [hunk 2]
```diff
@@ -261,56 +297,44 @@ contract RoundImplementation is AccessControlEnumerable, Initializable {
...
-    require(newApplicationsStartTime <= roundStartTime, "is after round start");
-    require(newApplicationsStartTime < applicationsEndTime, "is after app end");
-
-    emit ApplicationsStartTimeUpdated(applicationsStartTime, newApplicationsStartTime);
-
-    applicationsStartTime = newApplicationsStartTime;
-  }
+    require(newApplicationsStartTime < newApplicationsEndTime, "Round: Application end is before application start");
+    require(newRoundStartTime < newRoundEndTime, "Round: Round end is before round start");
+    require(newApplicationsStartTime < newRoundStartTime, "Round: Round start is before application start");
```
Comment (thelostone-mc) [inline packages/contracts/contracts/round/RoundImplementation.sol:315]:
```suggestion
    require(newApplicationsStartTime <= newRoundStartTime, "Round: Round start is before application start");
```

File: packages/contracts/contracts/round/RoundImplementation.sol [hunk 3]
```diff
@@ -261,56 +297,44 @@ contract RoundImplementation is AccessControlEnumerable, Initializable {
...
+    require(newApplicationsEndTime < newRoundEndTime, "Round: Round end is before application end");
```
Comment (thelostone-mc) [inline packages/contracts/contracts/round/RoundImplementation.sol:316]:
```suggestion
    require(newApplicationsEndTime <= newRoundEndTime, "Round: Round end is before application end");
```

File: packages/contracts/contracts/round/RoundImplementation.sol [hunk 4]
```diff
@@ -261,56 +297,44 @@ contract RoundImplementation is AccessControlEnumerable, Initializable {
...
+
+    if(newApplicationsStartTime > block.timestamp && applicationsStartTime > block.timestamp) {
```
Comment (thelostone-mc) [inline packages/contracts/contracts/round/RoundImplementation.sol:318]:
```suggestion
    if(newApplicationsStartTime != applicationsStartTime) {
```

Comment (thelostone-mc) [reply packages/contracts/contracts/round/RoundImplementation.sol:318]: Maybe we shift the block.timestamp check to one level above and do something like

```
require(newApplicationsStartTime >= block.timestamp
```

this check would be the only block.timestamp check needed for the entire function

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**